target/linux/ramips/image/mt7621.mk

```diff
@@ -506,6 +506,30 @@ define Device/netgear_r6850
 endef
 TARGET_DEVICES += netgear_r6850
 
+define Device/netgear_r6700v2
+  MTK_SOC := mt7621
+  DTS := R6700v2
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL_SIZE := 4096k
+  IMAGE_SIZE := 40960k
+  UBINIZE_OPTS := -E 5
+  SERCOMM_HWID := BZV
+  SERCOMM_HWVER := A001
+  SERCOMM_SWVER := 0x1032
+  IMAGES += factory.img kernel.bin rootfs.bin
+  IMAGE/factory.img := pad-extra 2048k | append-kernel | pad-to 6144k | append-ubi | \
+	pad-to $$$$(BLOCKSIZE) | sercom-footer | pad-to 128 | zip $$$$(DEVICE_MODEL).bin | sercom-seal
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  IMAGE/kernel.bin := append-kernel
+  IMAGE/rootfs.bin := append-ubi | check-size $$$$(IMAGE_SIZE)
+  DEVICE_VENDOR := NETGEAR
+  DEVICE_MODEL :=  R6700
+  DEVICE_VARIANT := v2
```

Comment:
This is a little tiresome with Netgear devices. The name generated from this will be "NETGEAR R6700 v2" (with space between version and model).
To be consistent, one would thus have to change model name in DTS accordingly, as well as change device node name to netgear_r6700-v2 and compatible to netgear,r6700-v2.
However, I have the feeling that Netgear typically uses the version without the space. If one followed that scheme, one would have to remove the DEVICE_VARIANT and use R6700V2 for DEVICE_MODEL, as the entire phrase is a "model name" then.

Since it effectively is a variant nevertheless, I have a (weak) preference for my first proposition, i.e. change device node/compatible.